plc4j/api/src/main/java/org/apache/plc4x/java/api/messages/specific/TypeSafePlcReadResponse.java

```diff
@@ -59,17 +59,49 @@ public TypeSafePlcReadResponse(TypeSafePlcReadRequest<T> request, List<ReadRespo
         return (Optional<ReadResponseItem<T>>) super.getResponseItem();
     }
 
+    /**
+     * Cast or convert a PlcReadResponse to a TypeSafePlcReadReadResponse<T>.
+     * 
+     * WARNING: this is inherently a non-type-safe operation.  It was introduced
+     * to serve the implementation of PlcReader.read(TypeSafePlcReadRequest<T>).
+     * Additional use of it is not recommended.  This interface is subject to change.
+     * 
+     * @param plcReadResponse the response implicitly with items of type T
+     * @return TypeSafePlcReadReadResponse<T>
+     */
     @SuppressWarnings("unchecked")
     public static <T> TypeSafePlcReadResponse<T> of(PlcReadResponse plcReadResponse) {
```

Comment:
fair Point, we would need a type Parameter `Class clazz` here.

Reply:
Seems like it.

Reply:
I fixed it in the latest master, so you might want to do a rebase/merge of master into your branch.